core/src/parking_lot.rs

```diff
@@ -5,19 +5,31 @@
 // http://opensource.org/licenses/MIT>, at your option. This file may not be
 // copied, modified, or distributed except according to those terms.
 
-use crate::thread_parker::ThreadParker;
-use crate::util::UncheckedOptionExt;
-use crate::word_lock::WordLock;
+use super::libstd::time::{Duration, Instant};
+use super::thread_parker::ThreadParker;
+use super::util::UncheckedOptionExt;
+use super::word_lock::WordLock;
 use core::{
     cell::{Cell, UnsafeCell},
     ptr,
     sync::atomic::{AtomicPtr, AtomicUsize, Ordering},
 };
+#[cfg(not(feature = "i-am-libstd"))]
 use rand::{rngs::SmallRng, FromEntropy, Rng};
+#[cfg(not(feature = "i-am-libstd"))]
 use smallvec::SmallVec;
-use std::time::{Duration, Instant};
 
+#[cfg_attr(all(test, feature = "i-am-libstd"), linkage = "available_externally")]
+#[cfg_attr(
+    feature = "i-am-libstd",
+    export_name = "_ZN3std16parking_lot_core11parking_lot11NUM_THREADSE"
+)]
 static NUM_THREADS: AtomicUsize = AtomicUsize::new(0);
+#[cfg_attr(all(test, feature = "i-am-libstd"), linkage = "available_externally")]
+#[cfg_attr(
+    feature = "i-am-libstd",
+    export_name = "_ZN3std16parking_lot_core11parking_lot9HASHTABLEE"
+)]
```

Comment:
Can you add a comment explaining why this is needed?

Reply:
Done.